game/Simulation.cpp

```diff
@@ -69,36 +69,32 @@ void Simulation::GetNextEvolution()
 	World* currentWorld = m_world;
 	World* nextWorld = new World(currentWorld->GetSize());
 
-	for (Player* player : m_players)
+	for (int i = 0; i < currentWorld->GetSize(); i++)
 	{
-		Coordinate currentLoc = player->GetLocation();
+		for (int j = 0; j < currentWorld->GetSize(); j++)
+		{
+			Coordinate currentLoc(i, j);
 
-		int numNeighbors = currentWorld->CountNeighbours(currentLoc);
+			int numNeighbors = currentWorld->CountNeighbours(currentLoc);
+			PlayerType newType = player->GetPlayerType();
```

Comment:
`player` is not defined here.  It is part of the old code u have which defined the player in foreach but not anymore.